dagger-common/src/main/java/com/gotocompany/dagger/common/core/StencilClientOrchestrator.java

```diff
@@ -37,13 +39,30 @@ public StencilClientOrchestrator(Configuration configuration) {
         this.stencilUrls = getStencilUrls();
     }
 
-    StencilConfig createStencilConfig() {
+    public StencilConfig createStencilConfig() {
         return StencilConfig.builder()
                 .fetchHeaders(getHeaders(configuration))
                 .fetchTimeoutMs(configuration.getInteger(SCHEMA_REGISTRY_STENCIL_FETCH_TIMEOUT_MS, SCHEMA_REGISTRY_STENCIL_FETCH_TIMEOUT_MS_DEFAULT))
+                .cacheAutoRefresh(configuration.getBoolean(SCHEMA_REGISTRY_STENCIL_CACHE_AUTO_REFRESH_KEY, SCHEMA_REGISTRY_STENCIL_CACHE_AUTO_REFRESH_DEFAULT))
+                .cacheTtlMs(configuration.getLong(SCHEMA_REGISTRY_STENCIL_CACHE_TTL_MS_KEY, SCHEMA_REGISTRY_STENCIL_CACHE_TTL_MS_DEFAULT))
+                .refreshStrategy(getSchemaRefreshStrategy(configuration.getString(SCHEMA_REGISTRY_STENCIL_REFRESH_STRATEGY_KEY, SCHEMA_REGISTRY_STENCIL_REFRESH_STRATEGY_DEFAULT)))
+                .fetchBackoffMinMs(configuration.getLong(SCHEMA_REGISTRY_STENCIL_FETCH_BACKOFF_MIN_MS_KEY, SCHEMA_REGISTRY_STENCIL_FETCH_BACKOFF_MIN_MS_DEFAULT))
+                .fetchRetries(configuration.getInteger(SCHEMA_REGISTRY_STENCIL_FETCH_RETRIES_KEY, SCHEMA_REGISTRY_STENCIL_FETCH_RETRIES_DEFAULT))
+                .updateListener(k -> TypeHandlerFactory.clearTypeHandlerMap())
```

Comment:
So I had 1 question. 

We are passing a lambda to `updateListener` here. Will the lambda be invoked after every `SCHEMA_REGISTRY_STENCIL_CACHE_TTL_MS_DEFAULT` duration, irrespective of whether the descriptor set has actually changed or not ? Or, will it only be invoked if there has been an actual update in the descriptor set ?

Reply:
Also, 1 scenario:

In our current Flink setup, when a job is submitted, the Job Manager requests for specific number of Task Managers to the Resource Manager as per the parallelism configured. It's possible that the task managers are registered one by one, in which case, there will be different `delays` at which stencil will be configured for schema fetch and refresh.

There can be a case where different TMs update their stencil cache at different intervals and hence the row that is transmitted by one TM is not parseable by the other TM ?

Reply:
> So I had 1 question.
> 
> We are passing a lambda to `updateListener` here. Will the lambda be invoked after every `SCHEMA_REGISTRY_STENCIL_CACHE_TTL_MS_DEFAULT` duration, irrespective of whether the descriptor set has actually changed or not ? Or, will it only be invoked if there has been an actual update in the descriptor set ?

it will be invoked after every cache ttl if the refresh strategy is LONG_POLLING and in case on VERSION_BASED_REFRESH , it will be invoked only if there's a new version of schema, CMIIW @harikrishnakanchi